etl/steps/data/garden/ihme_gbd/2019/gbd_tools.py

```diff
@@ -152,12 +153,177 @@ def add_share_of_population(df: pd.DataFrame) -> pd.DataFrame:
+            "description": "",
+            "unit": "cases",
+            "short_unit": "",
+            "num_decimal_places": 1,
+        },
+        "Prevalence - Rate": {
+            "title": f"Current number of cases of {cause.lower()} per 100,000 people, in {sex.lower()} aged {age.lower()}",
+            "description": "",
+            "unit": "cases",
+            "short_unit": "",
+            "num_decimal_places": 1,
+        },
+        "Incidence - Share of the population": {
+            "title": f"Number of new cases of {cause.lower()} per 100 people, in {sex.lower()} aged {age.lower()}",
+            "description": "",
+            "unit": "%",
+            "short_unit": "%",
+            "num_decimal_places": 1,
+        },
+        "Prevalence - Share of the population": {
+            "title": f"Current number of cases of {cause.lower()} per 100 people, in {sex.lower()} aged {age.lower()}",
+            "description": "",
+            "unit": "%",
+            "short_unit": "%",
+            "num_decimal_places": 1,
+        },
+    }
+    new_variable = variable.copy()
+    new_variable.name = underscore(var_name_dict[variable.name]["title"])
+    new_variable.metadata = VariableMeta(
+        title=var_name_dict[variable.name]["title"],
+        description=var_name_dict[variable.name]["description"],
+        unit=var_name_dict[variable.name]["unit"],
+        short_unit=var_name_dict[variable.name]["short_unit"],
+    )
+    new_variable.metadata.display = {
+        "name": var_name_dict[variable.name]["title"],
+        "numDecimalPlaces": var_name_dict[variable.name]["num_decimal_places"],
+    }
+
+    return new_variable
+
+
+def add_metadata_and_save(dest_dir: str, ds_meadow: Dataset, df: pd.DataFrame, dims: List[str]) -> None:
+    """
+    Adding metadata at the variable level
+    """
+    ds_garden = Dataset.create_empty(dest_dir)
+    ds_garden.metadata = ds_meadow.metadata
+
+    df = df.reset_index()
+    df_group = df.groupby(dims)
+    for group_id, group in df_group:
+        dims_id = dict(zip(dims, group_id))
+        cause = dims_id["cause"]
+        sex = dims_id["sex"]
+        age = dims_id["age"]
+        tb_group = Table(group)
+        tb_group.metadata.short_name = underscore(f"{dims_id['cause']} - {dims_id['sex']} - {dims_id['age']}")
+        variables = tb_group.columns.drop(dims + ["country", "year"])
+        for variable_name in variables:
+            tb_group[variable_name] = Variable(tb_group[variable_name])
+            cleaned_variable = create_variable_metadata(variable=tb_group[variable_name], cause=cause, sex=sex, age=age)
+            tb_group[cleaned_variable.name] = cleaned_variable
+            tb_group = tb_group.drop(columns=variable_name)
+        tb_group = tb_group.set_index(["country", "year"] + dims, verify_integrity=True)
+        ds_garden.add(tb_group)
+    ds_garden.save()
```

Comment:
I would change this function so it only adds the metadata and does not perform the save operation.

That is:

```
def add_metadata(dest_dir: str, ds_meadow: Dataset, df: pd.DataFrame, dims: List[str]) -> Dataset:
  ...
  return ds_garden
```

And then perform the `.save()` operation. Feels easier if we wanted to use the function to debug its output.